src/main.rs

```diff
@@ -737,6 +768,48 @@ fn backtrace(
             false
         };
 
+       // This is our first run through the loop, some initial handling and printing is required
+        // TODO refactor this
+        if frame_index == 0 {
+            if pc & !THUMB_BIT == vector_table.hard_fault & !THUMB_BIT {
+                // HardFaultTrampoline
+                // on hard fault exception entry we hit the breakpoint before the subroutine prelude (`push
+                // lr`) is executed so special handling is required
+                // also note that hard fault will always be the first frame we unwind
+
+                print_backtrace_start();
+
+                let mut stack_overflow = false;
+
+                if let Some(sp_ram_region) = sp_ram_region {
+                    // NOTE stack is full descending; meaning the stack pointer can be `ORIGIN(RAM) +
+                    // LENGTH(RAM)`
```

Comment:
```suggestion
                    // NOTE stack is full descending; meaning the stack pointer can be `ORIGIN(RAM) + LENGTH(RAM)`
```

Reply:
that would make the line longer than 100 chars (since I almost always look at 2 files simultaneously in vertical split view I tend to be very strict about this :D). I've changed the comment so that `ORIGIN(RAM) + LENGTH(RAM)` is all on one line though